version: "3.8"
services:
  database:
    image: mysql:8.0.15
    command: mysqld --default-authentication-plugin=mysql_native_password
    environment:
      - MYSQL_ROOT_PASSWORD=password
      - MYSQL_DATABASE=foody
    volumes:
      - ./database/schema.sql:/docker-entrypoint-initdb.d/1.schema.sql
      - ./database/content.sql:/docker-entrypoint-initdb.d/2.content.sql
    ports:
      - 4312:3306
    networks:
      - local
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      timeout: 20s
      retries: 10

  private-api:
    image: foody-private-api:latest
    command: "npm run start:dev"
    volumes:
      - ./private-api:/app/
    ports:
      - 5530:3000
    depends_on:
      database:
        condition: service_healthy
    networks:
      - local
      
  public-api:
    image: foody-public-api:latest
    command: "httpd -D FOREGROUND"
    volumes:
      - ./public-api:/app/
    ports:
      - "8055:8000"
    depends_on:
      - private-api
    networks:
      - local
      
  web-ui:
    image: foody-web-ui:latest
    command: "npm start"
    volumes:
      - ./web-ui:/app/
    ports:
      - "7755:3000"
    depends_on:
      - public-api
    networks:
      - local
      
networks:
  local:
